pkg/pod/status.go

```diff
@@ -128,7 +128,12 @@ func MakeTaskRunStatus(ctx context.Context, logger *zap.SugaredLogger, tr v1.Tas
 	complete := areStepsComplete(pod) || pod.Status.Phase == corev1.PodSucceeded || pod.Status.Phase == corev1.PodFailed
 
 	if complete {
-		updateCompletedTaskRunStatus(logger, trs, pod)
+		onError, ok := tr.Annotations[v1.PipelineTaskOnErrorAnnotation]
+		if ok {
+			updateCompletedTaskRunStatus(logger, trs, pod, v1.PipelineTaskOnErrorType(onError))
```

Comment:
Hi @JeromeJu. This is a good questions.

TEP-0050 is designed with requirement that `the task would be considered "successful" for the purposes of determining the status of the pipeline` (i.e. the `TaskRun` itself should still be failed, but with reason `FailureIgnored`). Looking at the [usecases](https://github.com/tektoncd/community/blob/main/teps/0050-ignore-task-failures.md#use-cases) in TEP-0050, it is mainly used to ignore pod error itself. 

I think your question is that for the validation/reconciler errors (like the `pvc creation`), should we also mark the reason as `FailureIgnored`? 

I think it might **not** be a good idea make the validation/reconciler errors  with reason `FailureIgnored`:
  - The validation/reconciler errors is likely from the Pipeline/Task Author, it shouldn't be ignored, but should be raised and fixed, `FailureIgnored` error reason could be confusing.
  - the error reason `FailureIgnored` becomes too generic if we use it across the TaskRun reconciler code. Instead, I'd prefer to keep the scope of `FailureIgnored` reason to the pod execution time.

WDYT?

Reply:
Thanks @QuanZhang-William for the offline conversation and completing the 2nd half of my question!

And yes, I do agree that in actual use cases, only the container(Step) failures in the pod, which indicates users' own test failures, shall be ignored but not the "preparation" of those steps i.e. pvc creation and validations. Thanks for the explanation here.

So I think the current implementation is the way to go. One small ask is maybe we would can add this explanation/conversation somewhere in the docString or maybe link to the TEP050 for this design choice.

Reply:
This makes sense as long as we document that `FailedIgnored` is ignoring a failure when the actual task execution fails but not otherwise (validation failure, etc.)

Reply:
SG. I have added the docstring. PTAL